eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/RouteBlinding.scala

```diff
@@ -34,9 +34,6 @@ sealed trait RouteBlindingEncryptedDataTlv extends Tlv
 
 object RouteBlindingEncryptedDataTlv {
 
-  /** Some padding can be added to ensure all payloads are the same size to improve privacy. */
-  case class Padding(dummy: ByteVector) extends RouteBlindingEncryptedDataTlv
```

Comment:
Wait why are we removing this? This is an official tlv in the specification, we should keep it?

Reply:
Reverted.
Actually we already create routes where all `encrypted_data`s have the same length, except for the last one. But I don't think it's a privacy leak that the last one has a different length, since the last one is already known to be for the sender. Dummy hops look just like regular hops without any need for padding.

Reply:
Thanks, now that the changes to route blinding are reverted, I'll let @remyers review the actual onion messages tests, and you can merge once you have an ACK.

> Actually we already create routes where all encrypted_datas have the same length, except for the last one. But I don't think it's a privacy leak that the last one has a different length, since the last one is already known to be for the recipient. Dummy hops look just like regular hops without any need for padding.

What I meant is that when we're adding dummy hops, we're _currently_ adding them *after* the hop that contains our final payload (and is thus larger than the other payloads). If we create a blinded path `Bob -> Us -> Dummy -> Dummy`, I believe the second payload will be larger than the others, which makes it obvious to the sender that the last two payloads are just dummy hops and we're actually one hop from Bob. That's where we'd use padding to remove this privacy leak.

But this will probably be an issue now that we're putting a lot of metadata in the blinded path to avoid storing the invoice? If we make all encrypted payloads the same size, this will likely overflow the payment onion since it's only 1300 bytes...so we should probably change our dummy hops implementation to always put our final payload in the last encrypted payload, and accept that this one will be larger (but it's ok as long as it's always the last one that is larger).

This is worth exploring in a separate PR though, I don't want to hijack this PR!

Reply:
> What I meant is that when we're adding dummy hops, we're _currently_ adding them _after_ the hop that contains our final payload (and is thus larger than the other payloads).

We're actually putting them _before_ the hop that contains our final payload. However I've just tested it and because we use a variable length encoding for the base fee (which is only for payments, so messages are not affected)  we can get smaller sizes for hops with a lower base fee and larger sizes for hops with a higher base fee which is a serious leak. If I remember correctly, the base fee was put at the end of the TLV with a variable length encoding so that it doesn't take any space in case of zero base fee but because of this leak it will force us to add a larger padding everywhere, maybe we should change the specs to use a fixed length encoding.